static/js/util/profile_edit.js

```diff
@@ -49,36 +55,66 @@ export function boundSelectField(keySet, label, options) {
     errors,
     updateProfile,
   } = this.props;
-  let onChange = value => {
+  let onChange = option => {
     let clone = _.cloneDeep(profile);
-    _.set(clone, keySet, value? value.value : '');
+    let toStore;
+    if (option.value !== null && option.value !== undefined) {
+      toStore = option.value.props.value;
```

Comment:
so here we're pulling the option value out of the `MenuItem` component we declare below (in `convertOption`). is there any reason we couldn't also stick a `value` prop on the return value of that function? it might make the code here a little simpler.